File: GoChat/src/main/java/com/jw/gochat/activity/FriendDetailActivity.java
package com.jw.gochat.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.jw.gochat.ChatApplication;
import com.jw.gochat.R;
import com.jw.gochat.base.BaseActivity;
import com.jw.gochat.bean.Account;
import com.jw.gochat.bean.Friend;
import com.jw.gochat.db.FriendDao;
import com.jw.gochat.view.NormalTopBar;

import Lib.GoChatURL;
import butterknife.BindView;

/**
 * 创建时间：2017/4/15
 * 更新时间 2017/10/29 2017/10/29
 * 版本：
 * 作者：Mr.jin
 * 描述：联系人详细信息
 */

public class FriendDetailActivity extends BaseActivity implements View.OnClickListener
        ,NormalTopBar.BackListener{

    @BindView(R.id.nt_friend_detail)
    NormalTopBar ntFriendDetail;
    @BindView(R.id.iv_list_item_friend_icon)
    de.hdodenhof.circleimageview.CircleImageView ivFriend;
    @BindView(R.id.tv_list_item_friend_name)
    TextView tvNameFriend;
    @BindView(R.id.tv_list_item_friend_act)
    TextView tvAccountFriend;
    @BindView(R.id.btn_friend_add)
    Button btnAdd;
    @BindView(R.id.btn_friend_send)
    Button btnSend;
    private Friend friend;
    private Account me = ChatApplication.getAccount();
    private FriendDao dao;

    @Override
    public void bindView() {
        setContentView(R.layout.activity_friend_detail);
    }

    @Override
    protected void initView() {
        super.initView();
        friend = (Friend) getIntent().getSerializableExtra("friend");
        dao = new FriendDao(FriendDetailActivity.this);
        if ((dao.queryFriendByAccount(me.getAccount(), friend.getAccount())) != null) {
            btnAdd.setVisibility(View.GONE);
            Glide.with(this).load(friend.getIcon()).into(ivFriend);
        } else {
            btnAdd.setVisibility(View.VISIBLE);
            Glide.with(this).load(GoChatURL.BASE_HTTP+friend.getIcon().replace("\\","/")).into(ivFriend);
        }
        tvNameFriend.setText(friend.getName());
        tvAccountFriend.setText(friend.getAccount());
    }

    @Override
    protected void initEvent() {
        super.initEvent();
        ntFriendDetail.setBackListener(this);
        btnAdd.setOnClickListener(this);
        btnSend.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btn_friend_add:
                finish();
                Intent intent1 = new Intent();
                intent1.putExtra("receiver", friend.getAccount());
                intent1.setClass(FriendDetailActivity.this, FriendValidateActivity.class);
                startActivity(intent1);
                break;
            case R.id.btn_friend_send:
                Intent intent2 = new Intent();
                Bundle bundle = new Bundle();
                bundle.putSerializable("receiver", friend);
                bundle.putParcelable("me", me);
                intent2.putExtras(bundle);
                intent2.setClass(FriendDetailActivity.this, MessageActivity.class);
                startActivity(intent2);
                break;
        }
    }

    @Override
    public void back() {
        finish();
    }
}

File: goChat/src/business/contact/java/com/jw/contact/activity/FriendDetailActivity.kt
package com.jw.contact

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.bumptech.glide.Glide
import com.jw.business.business.FriendBusiness
import com.jw.business.db.model.AccountInfo
import com.jw.business.db.model.Friend
import com.jw.chat.MessageActivity
import com.jw.gochat.GoChatApplication
import com.jw.gochat.R
import com.jw.gochat.databinding.ActivityFriendDetailBinding
import com.jw.gochat.view.NormalTopBar
import com.jw.gochatbase.gochat.GoChatURL
import com.jw.gochat.GoChatBindingActivity

/**
 * 创建时间：2017/4/15
 * 更新时间 2017/10/29 2017/10/29
 * 版本：
 * 作者：Mr.jin
 * 描述：联系人详细信息
 */

class FriendDetailActivity : GoChatBindingActivity<ActivityFriendDetailBinding>(), NormalTopBar.BackListener {
    private var friend: Friend? = null
    private lateinit var me: AccountInfo

    override fun getLayoutId() = R.layout.activity_friend_detail

    override fun doConfig(arguments: Intent) {
        val binding = binding
        me = GoChatApplication.getAccountInfo()!!
        friend = intent.getSerializableExtra("friend") as Friend
        binding.apply {
            if (FriendBusiness.getFriendById(me.account!!, friend!!.account!!) != null) {
                btnFriendAdd.visibility = View.GONE
                Glide.with(this@FriendDetailActivity).load(friend!!.icon).into(binding!!.ivListItemFriendIcon)
            } else {
                btnFriendAdd.visibility = View.VISIBLE
                Glide.with(this@FriendDetailActivity).load(GoChatURL.BASE_HTTP + friend!!.icon!!.replace("\\", "/")).into(ivListItemFriendIcon)
            }
            binding.tvListItemFriendName.text = friend!!.name
            binding.tvListItemFriendAct.text = friend!!.account
            ntFriendDetail.setBackListener(this@FriendDetailActivity)
            clickListener = View.OnClickListener {
                when (it.id) {
                    R.id.btn_friend_add -> {
                        finish()
                        val intent1 = Intent()
                        intent1.putExtra("receiver", friend!!.account)
                        intent1.setClass(this@FriendDetailActivity, FriendValidateActivity::class.java)
                        startActivity(intent1)
                    }
                    R.id.btn_friend_send -> {
                        val intent2 = Intent()
                        val bundle = Bundle()
                        bundle.putSerializable("receiver", friend)
                        bundle.putParcelable("me", me)
                        intent2.putExtras(bundle)
                        intent2.setClass(this@FriendDetailActivity, MessageActivity::class.java)
                        startActivity(intent2)
                    }
                }
            }
        }
    }

    override fun back() {
        finish()
    }
}